### DISCUSSION TITLE
Issue when using datasets slicing

### DISCUSSION THREAD

**Discussion by scottshufe** [pyg-team/pytorch_geometric#4112] [Q&A] [answered]
Hi pyg team,

I meet a problem when I use the dataset slicing, my codes (follow the pyg intro doc) :
```
from torch_geometric.datasets import TUDataset

dataset = TUDataset(root='/tmp/ENZYMES', name='ENZYMES')

data = dataset[5:10]
```

And the error info:
> Traceback (most recent call last):
>   File "D:/PycharmProjects/GCL_Privacy_Risk/Property inference attack/new.py", line 5, in <module>
>     data = dataset[5:10]
>   File "D:\ProgramData\Anaconda3\envs\GCL_risk\lib\site-packages\torch_geometric\data\dataset.py", line 203, in __getitem__
>     return self.index_select(idx)
>   File "D:\ProgramData\Anaconda3\envs\GCL_risk\lib\site-packages\torch_geometric\data\dataset.py", line 238, in index_select
>     dataset = copy.copy(self)
>   File "D:\ProgramData\Anaconda3\envs\GCL_risk\lib\copy.py", line 88, in copy
>     return copier(x)
> TypeError: __copy__() takes 1 positional argument but 2 were given

I'm using pytorch-geometric 2.0.2. I don't know what's wrong with my code. Could you please give me some suggestions to fix this issue?

**Comment by rusty1s** [comment, marked as answer]
I can't reproduce this issue on master or on `2.0.2`. It's a bit weird that your code complains about `copy.copy` receiving 2 arguments. Is your conda environment broken? What happens if you set up a new one?

**Reply by scottshufe** [reply]
Thanks for your quick reply. I'll check my env first.

**Reply by scottshufe** [reply]
I have changed my environment and the issue disappeared...It's confusing. Anyway, thanks for your help :)

**Reply by rusty1s** [reply]
Indeed weird. Let me know if you have any clue why this has happened :)

**Reply by scottshufe** [reply]
Okk.
